experimental/generation/generator/packages/library/src/testGenerator.ts

```diff
@@ -63,6 +63,12 @@ export async function generateTest(path: string, dialog: string, output: string,
             ++userCount
             responses = 0
             description = `Response to input ${userCount}`
+        } else if (isSettingRandomValue(record)) {
+            script.push({
+                $kind: 'Microsoft.Test.CustomEvent', 
+                name: 'SetTestOptions',
```

Comment:
>SetTestOptions [](start = 23, length = 14)

I like emitting the custom event since this uses the protocol and we know the underlying bot supports SetTestOptionsMiddleware to pick it up.